opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/Ctags.java

```diff
@@ -390,8 +410,43 @@ public Definitions doCtags(String file) throws IOException,
             if (Thread.interrupted()) {
                 throw new InterruptedException("flush()");
             }
-            readTags(rdr);
-            ret = rdr.getDefinitions();
+
+            /*
+             * Run the ctags reader in a time bound thread to make sure
+             * the ctags process completes so that the indexer can
+             * make progress instead of hanging the whole operation.
+             */
+            // TODO name the pool threads
+            ExecutorService executor = new ScheduledThreadPoolExecutor(1);
+            Future<Definitions> future = executor.submit(() -> {
+                Definitions defs = null;
+
+                try {
+                    readTags(rdr);
+                    defs = rdr.getDefinitions();
+                } catch (InterruptedException e) {
```

Comment:
Swallowing `InterruptedException` neutralizes retry-on-`InterruptedException` handling in `IndexDatabase`.

Reply:
Oh, I wondered what caused that, thanks !